src/model/json-helper.js

```diff
@@ -58,9 +58,17 @@ export function getNamespaceAndName(json={}, type) {
  */
 export function getNamespaceAndNameFromFHIR(fhir, type) {
   // Special case for primitives
-  if (typeof fhir !== 'object' && type != null && type.indexOf('.') === -1) {
-    return { namespace: 'primitive', elementName: type };
+  if (typeof fhir !== 'object') {
+    if (type == null) {
+      return { namespace: 'primitive', elementName: typeof fhir };
+    } else if (type.indexOf('.') === -1) {
+      return { namespace: 'primitive', elementName: type };
+    }
```

Comment:
I'm going to guess that this is a file that comes from the SHR toolchain but it seems like there is a condition that is missed here.  If typeof fhir !== 'object' and type.indexOf('.') !== -1 then nothing is returned.  This is probably a case where you would not expect the type to be some kind of object hierarchy so maybe an error should be thrown to prevent null pointers upstream.   

If this is an shr toolchain file then we can comment on that there.

Reply:
In the case where typeof fhir !== 'object' and type.indexOf('.') !== -1 then it will fall through to the logic below and eventually throw an error. I can make that more explicit with a separate branch or comment if you'd like.

This file does come from upstream in the shr toolchain and these changes should eventually make their way back up, but it was quicker to put them in place here first.

Reply:
Ok, I had some odd indentation issues where it looked like the method ended before it actually did.